Assets/Scripts/Inventory/Inventory.cs

```diff
@@ -2,65 +2,148 @@
 using System.Collections.Generic;
 using UnityEngine;
 
-// This class represents a slot within the player's inventory that holds an item
-[System.Serializable]
-public class ItemSlot
+namespace FunMath
 {
-    public ItemSlot(Item item, int Count)
+    // This class represents a slot within the player's inventory that holds an item
+    [System.Serializable]
+    public class ItemSlot
     {
-        this.item = item;
-        this.ItemCount = Count;
+        public ItemSlot(Item item, int Count)
+        {
+            this.item = item;
+            this.ItemCount = Count;
+        }
+        [SerializeField]
+        public Item item;
+        public int ItemCount = 0;
+        public bool Empty()
+        {
+            return ItemCount == 0;
+        }
     }
-    public Item item;
-    public int ItemCount = 0;
-}
-
-public class Inventory : MonoBehaviour
-{
-    public int MaxNumberOfItem = 100;
-    public bool InfiniteItem = false;
 
-    [SerializeField]
-    private List<ItemSlot> Items = new List<ItemSlot>(); // Holds the player's collected items
-    public List<ItemSlot> GetInventoryItemSlots()
+    public class Inventory : MonoBehaviour
     {
-        return Items;
-    }
+        public int MaxNumberOfItem = 100;
+        public bool InfiniteItem = false;

```

Comment:
Shall we move this as an item property?

Reply:
Yes, I wanted to do that before I had to leave but didn't finish it in time. There is a minor issue with Unity reflecting the right properties back, but I'll do this in another PR as I want to get stuff in ASAP.